### ISSUE TITLE
Classes vs Structs

### ISSUE THREAD

**Issue by Jebbs** [Jebbs/DSFML#148]
It seems to me that there could be some benefit in changing some classes to structs in DSFML.

Things that are POD or things that should be easy to copy might better be redefined as a struct instead of a class.

I'd like to go through and decide which should be which and make that change.

**Comment by aubade** [comment]
It's worth noting that struct destructors aren't guaranteed to be called ( https://issues.dlang.org/show_bug.cgi?id=2834 ), especially IIRC, if they're stored in an array. This is a bit dangerous when you have things constructed in C-space!

**Comment by Jebbs** [comment]
I found out that That issue you linked to has actually been fixed! Hooray!

**Comment by aubade** [comment]
Ooh, that's fantastic. Then yeah, I'm totally going to vote for View to get made into a struct :D 
I'm not sure what other classes would be good for structification, since a lot of the ones SFML has are either subclased, good for subclassing, or are pure references...

**Comment by Jebbs** [comment]
Yep. This isn't particularly high on my todo list right now. Once 2.1 is done (soon by the way) then I'll start thinking about it.

**Comment by Jebbs** [comment]
Superseded by #199
